libsolidity/ast/ASTJsonImporter.h

```diff
@@ -0,0 +1,118 @@
+/*
+    This file is part of solidity.
+
+    solidity is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    solidity is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author julius <djudju@protonmail.com>
+ * @date 2017
+ * Converts an AST from json format to an ASTNode
+ */
+
+#pragma once
+
+#include <vector>
+#include <libsolidity/ast/AST.h>
+#include <json/json.h>
+#include <libsolidity/ast/ASTAnnotations.h>
+#include <libevmasm/SourceLocation.h>
+
+namespace dev
+{
+namespace solidity
+{
+
+/**
+ * takes an AST in Json Format and recreates it with AST-Nodes
+ */
+class ASTJsonImporter
+{
+public:
+	/// Create an Importer to import a given abstract syntax tree in Json format to an ASTNode
+	ASTJsonImporter(std::map<std::string, Json::Value const*> _sourceList);
+
+        ///this is the primary Interface from the Outside
```

Comment:
Would it be better to have just a single static function and a private constructor? Is there a benefit of having a two-step process?